scripts/lib/CIME/utils.py

```diff
@@ -660,6 +660,11 @@ def handle_standard_logging_options(args):
     stderr_stream_handler = logging.StreamHandler(stream=sys.stderr)
     stderr_stream_handler.setLevel(logging.WARNING)
 
+    # scripts_regression_tests is the only thing that should pass a None argument in parser
+    if parser is not None:
+        _check_for_invalid_args(args[1:])
+        args = parser.parse_args(args[1:])
```

Comment:
If we're going to rely on this function to do the actual parsing of arguments, then it should be renamed. Currently, as I'm reading through the code, I am left wondering where the actual argument parsing happens: it is not intuitive that the argument parsing would happen inside a function named 'handle_standard_logging_options'. Maybe even rename this function to 'parse_args'?